core/src/avm2/object/loaderinfo_object.rs

```diff
@@ -177,14 +183,29 @@ impl<'gc> LoaderInfoObject<'gc> {
                     e
                 );
             }
+        }
 
-            let complete_evt = EventObject::bare_default_event(context, "complete");
-
-            if let Err(e) = Avm2::dispatch_event(context, complete_evt, (*self).into()) {
-                log::error!(
-                    "Encountered AVM2 error when broadcasting `complete` event: {}",
-                    e
-                );
+        if !self.0.read().complete_event_fired {
+            // NOTE: We have to check load progress here because this function
+            // is called unconditionally at the end of every frame.
+            let should_complete = match self.0.read().loaded_stream {
+                Some(LoaderStream::Swf(_, root)) => root
+                    .as_movie_clip()
+                    .map(|mc| mc.loaded_bytes() >= mc.total_bytes())
```

Comment:
Can we ever have a movie where `loaded_bytes` is greater than `total_bytes`?

Reply:
We shouldn't, this is just paranoia on my part.